test/units/module_utils/common/text/formatters/test_lenient_lowercase.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019, Andrew Klychkov @Andersson007 <aaklychkov@mail.ru>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.common.text.formatters import lenient_lowercase
+
+
+INPUT_LIST = [
+    u'HELLO',
+    u'Ёлка',
+    u'cafÉ',
+    b'HELLO',
+    1,
+    {1: 'Dict'},
+    True,
+    [1],
+]
+
+LOWERED_LIST = [
+    u'hello',
+    u'ёлка',
+    u'café',
+    b'hello',
+    1,
+    {1: 'Dict'},
+    True,
+    [1],
+]
+
+
+def test_lenient_lowercase():
```

Comment:
Please use `pytest.mark.parametrize()` instead of a loop

Reply:
@webknjaz , IMO pytest.mark.parametrize is not suitable for this particular case, I can't imagine how to implement this.
The input list is passed once, then we just compare each element of it with a corresponding element of the expected list

Reply:
You mess up separate test cases into one. While they should show up separately. Also, one failure shouldn't affect other tests.

Reply:
```suggestion
@pytest.mark.parametrize(
    'input_value,expected_outcome',
    (
        (i1, r1),
        (i2, r2),
       ...etc
    ),
)
def test_lenient_lowercase(input_value, expected_outcome):
```

Reply:
thank you, PTAL